crypto/rsa/rsa_sp800_56b_gen.c

```diff
@@ -392,6 +393,13 @@ int ossl_rsa_sp800_56b_generate_key(RSA *rsa, int nbits, const BIGNUM *efixed,
         /* (Step 2) Generate prime factors */
         if (!ossl_rsa_fips186_4_gen_prob_primes(rsa, info, nbits, e, ctx, cb))
             goto err;
+
+        if ((BN_cmp(rsa->p, rsa->q) < 0) && !info) {
```

Comment:
Please do the `info` test before the `BN_cmp()` test, as it is much faster

Reply:
Also, we generally do `== NULL` rather than `!info` for pointers (style).